<a href="https://colab.research.google.com/github/Sandramcrs08/Teste_de_Hipoteses_Experimento/blob/main/Estat%C3%ADstica_Teste_de_Hip%C3%B3teses.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Estatística- Teste de Hipóteses Considerando a base de dados experimento_test_ab.csv responda as questões abaixo:


In [ ]:
!pip install pandas
import pandas as pd
import numpy as np
from statsmodels.stats.proportion import proportions_ztest
from scipy.stats import norm


In [ ]:
# Carregar os dados
dados = pd.read_csv('/experimento_test_ab.csv')

# Visualizar os dados
print(dados.head())


   Visitante_ID Versão_Página  Visualizações  Conversões
0             1             A            167           1
1             2             A            160           0
2             3             A            170           0
3             4             A            161           1
4             5             A            104           0


In [ ]:
print(dados.columns)

Index(['Visitante_ID', 'Versão_Página', 'Visualizações', 'Conversões'], dtype='object')


1. Qual dos cenários tem a maior taxa de conversão?

In [ ]:
# Calcular a taxa de conversão para cada cenário
conversao_por_cenario = dados.groupby('Versão_Página')['Conversões'].mean()
print("\nTaxa de Conversão por Cenário:")
print(conversao_por_cenario)

# Identificar o cenário com a maior taxa de conversão
maior_taxa_conversao = conversao_por_cenario.idxmax()
print("\nCenário com a Maior Taxa de Conversão:", maior_taxa_conversao)


Taxa de Conversão por Cenário:
Versão_Página
A    0.105
B    0.125
Name: Conversões, dtype: float64

Cenário com a Maior Taxa de Conversão: B


2. Calcule qual o tamanho da amostra necessária para o desenvolvimento de um teste A/B, seguindo os seguintes critérios:

a. O cenário A, da base, como o inicial, que funciona hoje.

In [ ]:
# Taxa de conversão para o cenário A (base)
taxa_conversao_cenario_A = 0.08


R: A taxa de conversão atual para o cenário A (8%).

b. Considere 95% de confiança de que o efeito na conversão não foi aleatório com um nível de significância de 5% (alpha = 0,05).

In [ ]:
# Nível de confiança (95%)
confianca = 0.95

# Nível de significância (alpha)
alpha = 0.05

# Z-score para o nível de confiança
Z_confianca = norm.ppf(confianca)

# Z-score para o poder do teste
Z_poder = norm.ppf(1 - alpha/2)

# Calculando o tamanho da amostra necessário para o teste A/B
n = (Z_confianca + Z_poder)**2

# Arredondando para o próximo número inteiro
n = np.ceil(n)

print("Tamanho da amostra necessário para o teste A/B:", int(n))


Tamanho da amostra necessário para o teste A/B: 13


R: Tamanho da amostra necessário para o teste A/B: 13

c. Também considere 80% de certeza conseguir capturar o efeito da nova abordagem.

In [ ]:
# Nível de confiança (95%)
confianca = 0.95

# Nível de significância (alpha)
alpha = 0.05

# Poder do teste (80%)
poder = 0.8

# Z-score para o nível de confiança
Z_confianca = norm.ppf(confianca)

# Z-score para o poder do teste
Z_poder = norm.ppf(poder)

# Calculando o tamanho da amostra necessário para o teste A/B
n = (Z_confianca + Z_poder)**2

# Arredondando para o próximo número inteiro
n = np.ceil(n)

print("Tamanho da amostra necessário para o teste A/B com 80% de poder:", int(n))


Tamanho da amostra necessário para o teste A/B com 80% de poder: 7


R: Tamanho da amostra necessário para o teste A/B com 80% de poder: 7

d. O aumento para 10% de conversão.

In [ ]:
# Taxa de conversão no cenário A (base)
taxa_A = 0.10

# Taxa de conversão no cenário B (nova abordagem)
taxa_B = 0.10 + 0.10  # Aumento para 10% de conversão

# Diferença mínima detectável entre as taxas de conversão
dif_min_detectavel = taxa_B - taxa_A

# Nível de confiança (95%)
confianca = 0.95

# Nível de significância (alpha)
alpha = 0.05

# Poder do teste (80%)
poder = 0.8

# Z-score para o nível de confiança
Z_confianca = norm.ppf(confianca)

# Z-score para o poder do teste
Z_poder = norm.ppf(poder)

# Calculando o tamanho da amostra necessário para o teste A/B
n = ((Z_confianca + Z_poder)**2) * ((taxa_A * (1 - taxa_A)) + (taxa_B * (1 - taxa_B))) / (dif_min_detectavel**2)

# Arredondando para o próximo número inteiro
n = np.ceil(n)

print("Tamanho da amostra necessário para o teste A/B com aumento para 10% de conversão:", int(n))


Tamanho da amostra necessário para o teste A/B com aumento para 10% de conversão: 155


R: Tamanho da amostra necessário para o teste A/B com aumento para 10% de conversão: 155.

Verifique a base pacientes.csv, os dados são fictícios. A sintetiza uma base de dados de um hospital que trata de pacientes com problemas cardíacos. Considere a base como a população, portanto as estatísticas da população são conhecidas. Responda as perguntas abaixo:

3. Considerando uma amostra de 45 números que representam o index do dataframe, índices= ([909, 751, 402, 400, 726, 39, 184, 269, 255, 769, 209, 715, 677, 381, 793, 697, 89, 280, 232, 756, 358, 36, 439, 768, 967, 699, 473, 222, 89, 639, 883, 558, 757, 84, 907, 895, 217, 224, 311, 348, 146, 505, 273, 957, 362]). Considerando essa amostra é possível dizer que a idade média das pessoas com problemas cardíacos é maior que 50 anos? Nível de significância igual a 5%.

In [ ]:
# Carregar o arquivo pacientes.csv em um DataFrame
df = pd.read_csv('/content/drive/MyDrive/pacientes.csv')

# Visualizar os dados
print(df.head())

In [ ]:
# Índices da amostra
indices_amostra = [909, 751, 402, 400, 726, 39, 184, 269, 255, 769, 209, 715, 677, 381, 793, 697, 89, 280, 232, 756, 358, 36, 439, 768, 967, 699, 473, 222, 89, 639, 883, 558, 757, 84, 907, 895, 217, 224, 311, 348, 146, 505, 273, 957, 362]

# Selecionar as idades das pessoas com problemas cardíacos
idades_pessoas_com_problemas_cardiacos = df.loc[indices_amostra, 'Idade']

idades_pessoas_com_problemas_cardiacos.head()

In [ ]:
# Calcular a média das idades
media_idades = np.mean(idades_pessoas_com_problemas_cardiacos)
print("Média das idades:", media_idades)


In [ ]:
# Nível de significância
nivel_significancia = 0.05

# Número de observações na amostra
n = len(idades_pessoas_com_problemas_cardiacos)

# Graus de liberdade
graus_liberdade = n - 1

# Estatística de teste t
t_valor = (media_idades - 50) / (np.std(idades_pessoas_com_problemas_cardiacos, ddof=1) / np.sqrt(n))



In [ ]:
# Cálculo do p-valor (usando a distribuição t de Student)
from scipy.stats import t
p_valor = 1 - t.cdf(t_valor, graus_liberdade)

# Imprimir os resultados
print("Estatística de teste t:", t_valor)
print("p-valor:", p_valor)

# Comparar o p-valor com o nível de significância
if p_valor < nivel_significancia:
    print("Rejeitar a hipótese nula")
    print("A idade média das pessoas com problemas cardíacos é estatisticamente maior que 50 anos.")
else:
    print("Não rejeitar a hipótese nula")
    print("Não há evidências suficientes para afirmar que a idade média das pessoas com problemas cardíacos é maior que 50 anos.")
    # Esse código calculará corretamente o p-valor.

Com um nível de significância de 0.05, não há evidências suficientes para afirmar que a idade média das pessoas com problemas cardíacos é maior que 50 anos.

4. Queremos entender que tipo de amostra estamos lidando se dividirmos os conjuntos em 2, sendo um com pessoas que têm condições de saúde adicionais e o outro com pessoas saudáveis. Seria dependente ou independente?

In [ ]:
print(df.columns)

In [ ]:
# Dividindo o DataFrame em dois grupos com base no estado de saúde
grupo_saude_adicional = df[df['Estado_Saude'] == "Com condições de saúde adicionais"]
grupo_saudavel = df[df['Estado_Saude'] == "Saudável"]

# Verificando se as amostras são dependentes ou independentes
# Teste t de Student para amostras independentes
t_statistic, p_valor = stats.ttest_ind(grupo_saude_adicional, grupo_saudavel)

# Definindo um nível de significância
nivel_significancia = 0.05

# Verificando se as amostras são dependentes ou independentes com base no valor-p
if p_valor.any() < nivel_significancia:
    print("Com um nível de significância de", nivel_significancia)
    print("As amostras são independentes.")
else:
    print("Com um nível de significância de", nivel_significancia)
    print("As amostras não são independentes.")

# Comparando o valor-p para verificar a significância estatística
if p_valor.any() < nivel_significancia:
    print("O teste t de Student para amostras independentes é significativo (valor-p =", p_valor, ")")
else:
    print("O teste t de Student para amostras independentes não é significativo (valor-p =", p_valor, ")")